### DISCUSSION TITLE
Allow to request information about multiple model ids using REST API

### DISCUSSION THREAD

**Discussion by db0** [civitai/civitai#557] [General]
I have a list of model IDs I want to fetch like so

```json
[
    48139,
    25995,
    16014,
    51686,
    51145,
    44960
 ]
 ```
    
Normally I would need to make one request per model @ [GET /api/v1/models/:modelId](https://github.com/civitai/civitai/wiki/REST-API-Reference#get-apiv1modelsmodelid)
I would like instead to request a way to retrieve them all in one go, by passing them to [GET /api/v1/models](https://github.com/civitai/civitai/wiki/REST-API-Reference#get-apiv1models) instead. 

Perhaps a comma-separated list with a specific query like

`https://civitai.com/api/v1/models?ids=48139,25995`

This would serve to reduce load on your servers and to get my responses faster

**Comment by JustMaier** [comment]
Yeah, that seems convenient. I'll get this added to the todo list.

**Comment by bkdiehl** [comment]
This is ready to go. One important distinction is that you'll need to format your query string array like this:
```
/api/v1/models?ids=48139&ids=25995&ids=...etc
```

**Comment by db0** [comment]
Got it! 

PS: But weird to call it "ids" plural and specify only 1 though :D

**Comment by BlafKing** [comment]
Glad I stumbled upon this!

This will greatly improve the speed in which I can check for available updates in my A1111 extension by doing a single API request.
Is something similar for hashes perhaps planned/available?
